cia.mk

```diff
@@ -1,59 +1,91 @@
 ### Virtual Console repacking stuff
 
-# Include Configuration Settings
 vc_dir := poke-cia
--include $(vc_dir)/cia-config.mk
+
+# Include Configuration Settings
+include ${vc_dir}/cia-config.mk
+ifeq ($(strip ${vc_name}),)
+$(error Please set the `vc_name` variable in ${vc_dir}/cia-config.mk)
+endif
+
+vc_rom_dirs   := $(addprefix ${vc_dir}/, ${vc_name})
+vc_cias       := $(addsuffix .cia, ${vc_rom_dirs})
+vc_orig_cia   := $(addsuffix .orig.cia, ${vc_rom_dirs})
+vc_game_cxi   := $(addsuffix .game.cxi, ${vc_rom_dirs})
+vc_manual_cfa := $(addsuffix .manual.cfa, ${vc_rom_dirs})
+
+
+# "Interface" rules
 
 .PHONY: cia
-cia: $(vc_cia)
+cia: ${vc_cias}
 
 .PHONY: clean
 clean: clean_cia
 
 .PHONY: clean_cia
 clean_cia:
-	rm -f $(vc_cia) $(vc_game_cxi) $(vc_manual_cfa)
+	rm -f ${vc_cias} ${vc_game_cxi} ${vc_manual_cfa}
 
 .PHONY: distclean_cia
 distclean_cia: clean_cia
-	rm -rf $(vc_rom_dir) $(vc_dir)/seeddb.bin
+	rm -rf ${vc_rom_dirs} ${vc_dir}/seeddb.bin
+
 
-$(vc_rom_dir): $(vc_orig_cia) $(vc_dir)/seeddb.bin
+# Actual rules
+
+# TODO: relying on directory modification time is a bad idea!
+# Silence `ctrtool`, which is VERY verbose by default
+${vc_dir}/%/: ${vc_dir}/%.orig.cia ${vc_dir}/seeddb.bin
 	mkdir -p $@
-	ctrtool --contents=$@/contents $@.orig.cia
-	ctrtool --seeddb=$(vc_dir)/seeddb.bin \
-	        --exheader=$@/exheader.bin \
-	        --exefsdir=$@/exefs \
-	        --romfsdir=$@/romfs \
-	        --logo=$@/logo.lz \
-	        --plainrgn=$@/plain.bin \
-	        $@/contents.0000.*
-	ctrtool --seeddb=$(vc_dir)/seeddb.bin \
-	        --romfsdir=$@/manual \
-	        $@/contents.0001.*
-	rm -f $@/contents.*
-	rm -f $@/romfs/rom/*
-	rm -f $@/romfs/*.patch
-
-$(join $(addsuffix /romfs/, $(vc_rom_dir)), $(vc_patch)): $(vc_rom_dir) ./$(vc_patch)
-	cp ./$(@F) $@
-	
-$(join $(addsuffix /romfs/rom/, $(vc_rom_dir)), $(vc_gbc)): $(vc_rom_dir) ./$(vc_gbc)
-	cp ./$(@F) $(basename $@)
-
-$(vc_game_cxi): $(vc_rom_dir) $(join $(addsuffix /romfs/, $(vc_rom_dir)), $(vc_patch)) $(join $(addsuffix /romfs/rom/, $(vc_rom_dir)), $(vc_gbc))
-	(cd $(basename $(basename $@))/; \
-	    makerom -f cxi -o ../../$(basename $@).cxi -rsf ../game.rsf \
-	            -exheader exheader.bin \
-	            -logo logo.lz \
-	            -plainrgn plain.bin \
-	            -code exefs/code.bin \
-	            -icon exefs/icon.bin \
-	            -banner exefs/banner.bin \
-	)
-
-$(vc_manual_cfa): $(vc_dir)/manual.rsf
-	makerom -f cfa -o $@ -rsf $(vc_dir)/manual.rsf
-
-$(vc_cia): $(vc_game_cxi) $(vc_manual_cfa)
-	makerom -f cia -o $@ -content $(basename $@).game.cxi:0:0 -content $(basename $@).manual.cfa:1:1
\ No newline at end of file
+	ctrtool --contents=$@contents $< >/dev/null
```

Comment:
if you are using a CTRTool v1.0.0 and newer, you don't have to extract the CIA to process the contents.

Instead of doing this
```
	ctrtool --contents=$@contents $< >/dev/null
	ctrtool --seeddb=${vc_dir}/seeddb.bin \
 	        --exheader=$@exheader.bin \
 	        --exefsdir=$@exefs \
 	        --romfsdir=$@romfs \
 	        --logo=$@logo.lz \
 	        --plainrgn=$@plain.bin \
 	        $@contents.0000.* >/dev/null
	ctrtool --seeddb=${vc_dir}/seeddb.bin \
 	        --romfsdir=$@manual \
 	        $@contents.0001.* >/dev/null
```
Try:
```
	ctrtool --cidx 0 \
 	        --seeddb=${vc_dir}/seeddb.bin \
 	        --exheader=$@exheader.bin \
 	        --exefsdir=$@exefs \
 	        --romfsdir=$@romfs \
 	        --logo=$@logo.lz \
 	        --plainrgn=$@plain.bin \
 	        $< >/dev/null
	ctrtool --cidx 1 \
 	        --seeddb=${vc_dir}/seeddb.bin \
 	        --romfsdir=$@manual \
 	        $< >/dev/null
```

Reply:
Thanks for the tip! That worked for me after I updated my ctrtool XD.

Reply:
Great! I'm using 1.0.1, so I'll try that. We'll have to note that minimum version required—I'll mark 1.0.0, as it's a patch-level difference.